src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/RouteRequestTransitDataProviderFilter.java

```diff
@@ -143,17 +144,39 @@ public boolean tripTimesPredicate(TripTimes tripTimes, boolean withFilters) {
   }
 
   @Override
-  public BitSet filterAvailableStops(RoutingTripPattern tripPattern, BitSet boardingPossible) {
+  public BitSet filterAvailableStops(
+    RoutingTripPattern tripPattern,
+    BitSet boardingPossible,
+    BoardAlight boardAlight
+  ) {
+    var result = boardingPossible;
+
+    // if the user wants to include realtime cancellations, include cancelled stops as available
+    if (includeRealtimeCancellations) {
+      var pattern = tripPattern.getPattern();
+      var nStops = pattern.numberOfStops();
+      result = new BitSet(nStops);
+
+      for (int i = 0; i < nStops; i++) {
+        PickDrop pickDrop =
+          switch (boardAlight) {
```

Comment:
If we want to shave of a few cycles of CPU time the choosing of getter method can be decided outside of the loop and stored as a lambda expression. Doubtful if would have a measurable effect though, code is cheap.